src/Nethermind/Nethermind.Trie/TrieNode.cs

```diff
@@ -111,6 +116,50 @@ void ThrowAlreadySealed()
                 }
             }
         }
+        public ref readonly CappedArray<byte> ValueRef
+        {
+            get
+            {
+                InitData();
+                if (IsLeaf)
+                {
+                    object? data = _data![1];
+                    if (data is null)
+                    {
+                        return ref CappedArray<byte>.Null;
+                    }
+
+                    return ref Unsafe.Unbox<CappedArray<byte>>(data);
+                }
+
+                if (!AllowBranchValues)
+                {
+                    // branches that we use for state will never have value set as all the keys are equal length
+                    return ref CappedArray<byte>.Empty;
+                }
+
+                ref object? obj = ref _data![BranchesCount];
+                if (obj is null)
+                {
+                    RlpFactory rlp = _rlp;
+                    if (rlp is null)
+                    {
+                        obj = CappedArray<byte>.EmptyBoxed;
+                        return ref CappedArray<byte>.Empty;
+                    }
+                    else
+                    {
+                        ValueRlpStream rlpStream = rlp.GetRlpStream();
+                        SeekChild(ref rlpStream, BranchesCount);
+                        byte[]? bArr = rlpStream.DecodeByteArray();
+                        obj = new CappedArray<byte>(bArr);
+                        return ref Unsafe.Unbox<CappedArray<byte>>(obj);
```

Comment:
Do we need to box and unbox later?

Reply:
Object is `ref object` so is boxed; but also need to attach it to something to pass back the ref to. Can't pass back the stack reference as it would become invalidated as the stack unwinds (also C# won't let us)